nifi-commons/nifi-properties/src/main/java/org/apache/nifi/util/NiFiProperties.java

```diff
@@ -303,6 +303,18 @@
     public static final String MONITOR_LONG_RUNNING_TASK_SCHEDULE = "nifi.monitor.long.running.task.schedule";
     public static final String MONITOR_LONG_RUNNING_TASK_THRESHOLD = "nifi.monitor.long.running.task.threshold";
 
+    // automatic diagnostic properties
+    public static final String AUTOMATIC_DIAGNOSTIC_ALLOWED = "nifi.auto.diag.allowed";
+    public static final String AUTOMATIC_DIAGNOSTIC_VERBOSE = "nifi.auto.diag.verbose";
+    public static final String AUTOMATIC_DIAGNOSTIC_DIRECTORY = "nifi.auto.diag.dir";
+    public static final String AUTOMATIC_DIAGNOSTIC_MAX_FILE_COUNT = "nifi.auto.diag.filecount.max";
+    public static final String AUTOMATIC_DIAGNOSTIC_MAX_DIR_SIZE_IN_BYTES = "nifi.auto.diag.dir.size.max.byte";
```

Comment:
These property names do not match the names in the documentation or example `nifi.properties` configuration.

It would be helpful to name the properties using the general category first, as well as spelling out the property name so that it matches the static variable name convention.

For example: `nifi.diagnostics.enabled` or `nifi.diagnostics.on.shutdown.enabled`.  Perhaps `nifi.diagnostics.collection.enabled`, either of which might be clearer than `nifi.diagnostics.automatic.enabled`.

Reply:
Thanks for the tip.